src/app/containers/RadioSchedule/Canonical/useData.jsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect, useContext } from 'react';
+import { RequestContext } from '#contexts/RequestContext';
+import webLogger from '#lib/logger.web';
+
+const logger = webLogger();
+
+// we'd put this in a hooks folder, not with radio schedules
+const useData = endpoint => {
+  const { ssrData } = useContext(RequestContext); // we would put this in a different context probably
+  const initialData = ssrData[endpoint] && ssrData[endpoint].json;
+  const [data, setData] = useState(initialData);
+
+  useEffect(() => {
+    const handleResponse = async response => {
+      const json = await response.json();
+      setData(json);
+    };
+
+    const fetchData = pathname =>
+      fetch(pathname, { mode: 'no-cors' })
+        .then(handleResponse)
+        .catch(e => logger.error(`HTTP Error: "${e}"`));
+
+    if (!data) {
+      fetchData(endpoint);
+    }
+
+    return () => {
+      // remove the initial data from context at this point
+    };
+  }, [endpoint, ssrData, data]);
```

Comment:
Double check... if I update data inside this useEffect, but also rely on the value of it (i.e., include it in the dependencies array), i _shouldn't_ go into an infinite loop because useEffect will only run again if the value of `data` changes. Make sure this works as expected and we don't make multiple API calls

Reply:
Every time I work with useEffect hooks I think I understand them and then question myself next time I use them

Reply:
OK. So it goes like this:

- Outside useEffect: data is undefined
- Inside useEffect: data is undefined
   We call setData inside useEffect which updates the state.

RERENDER

- Outside useEffect: data is our json payload
data has changed, so the useEffect runs again.
- Inside useEffect: data is defined, so we no longer fetch the data.

Now the useEffect only runs again if `data` is updated from a different place (or the endpoint or ssrData changes).